components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/RoleBasedInternalScopeValidator.java

```diff
@@ -157,7 +157,8 @@
             if (userName == null) {
                 userName = userStoreManager.getUserNameFromUserID(authenticatedUser.getUserId());
             }
-            return userStoreManager.getHybridRoleListOfUser(userName, authenticatedUser.getUserStoreDomain());
+            return userStoreManager.getHybridRoleListOfUser(userName.toLowerCase(),
```

Comment:
Is this fix valid in cases where the username is case sensitive?

To me if the username is case sensitive, the authentication should fail. Passing a lower cases username should only be done if the username is treated as **case insensitive** (This is usually checked via a config)

Reply:
+1. This is addressed.